.github/workflows/test-with-pypi.yml

```diff
@@ -0,0 +1,71 @@
+name: Test with PyPI
+
+on: [pull_request, workflow_dispatch]
+
+env:
+  PYTHONUNBUFFERED: 1
+
+jobs:
+
+  # Test against PyPI packages
```

Comment:
Do you mean why separate into ubuntu and others? I think the installation command is different (sudo apt-get install), also we might want to test with different python versions on different platforms.

Reply:
Yes, that's what I meant. There are trade-offs here; on one hand, having two separate jobs that do the same thing is a bit of a violation of DRY. On the other hand, it may make sense if there are sufficient differences. I leave it to you.

For the apt-get instructions, it's easy to make that step conditional so that it only runs on Ubuntu systems.